src/features/aptosSignAndSubmitTransaction.ts

```diff
@@ -24,6 +25,12 @@ export type AptosSignAndSubmitTransactionMethod = (
   transaction: AptosSignAndSubmitTransactionInput
 ) => Promise<UserResponse<AptosSignAndSubmitTransactionOutput>>
 
-export type AptosSignAndSubmitTransactionInput = AnyRawTransaction
-/** Output of signing transactions. */
-export type AptosSignAndSubmitTransactionOutput = PendingTransactionResponse
+export interface AptosSignAndSubmitTransactionInput {
+  gasUnitPrice?: number;  // defaults to estimated gas unit price
+  maxGasAmount?: number;  // defaults to estimated max gas amount
+  payload: TransactionPayload | InputGenerateTransactionPayloadData;
+}
+
+export interface AptosSignAndSubmitTransactionOutput {
```

Comment:
interesting, why this change? we might dont want to enforce the dapp to have the wallet waits for the transaction to be submitted?

Reply:
We don't really need the whole `PendingTransactionResponse` object back typically.
The hash is all the dapp needs to fetch the status and was what we returned previously.
It's always easier to add more parameters later if they're required.

Reply:
I wouldn't be opposed to keeping `PendingTransactionResponse`, but it can be annoying i.e. for Petra we would have to change the code so that the additional fields are sent together with the `hash`